.github/workflows/inventory.yml

```diff
@@ -7,14 +7,54 @@ on:
 
 jobs:
   update-nodejs-inventory:
-    name: Node.js
-    uses: ./.github/workflows/_update-inventory.yml
-    with:
-      name: Node.js
-      distribution: node
-      buildpack_id: heroku/nodejs-engine
-      buildpack_path: buildpacks/nodejs-engine
-    secrets: inherit
+    name: Update Node.js Inventory
+    runs-on: pub-hk-ubuntu-22.04-small
+    steps:
+      - uses: heroku/use-app-token-action@main
+        id: generate-token
+        with:
+          app_id: ${{ vars.LINGUIST_GH_APP_ID }}
+          private_key: ${{ secrets.LINGUIST_GH_PRIVATE_KEY }}
+
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+
+      - id: install-rust-toolchain
+        name: Install Rust Toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+
+      - name: Rebuild Inventory
+        id: rebuild-inventory
+        run: |
+          delimiter="$(openssl rand -hex 8)"
+          {
+            echo "msg<<${delimiter}"
+            cargo run --bin update_node_inventory buildpacks/nodejs-engine/inventory.toml
+            echo "${delimiter}"
+          } >> $GITHUB_OUTPUT
+
+      - name: Update Changelog
+        run: echo "${{ steps.rebuild-inventory.outputs.msg }}" | xargs -r -I '{}' perl -i -p -e 's/\[Unreleased\]\s+/[Unreleased]\n\n- {}/' buildpacks/nodejs-engine/CHANGELOG.md
```

Comment:
Since you're in here, maybe now is a good time to leverage `heroku/keep_a_changelog` since this search-and-replace doesn't conform to the Keep a Changelog spec ;)

Reply:
I'd be delighted to do this :-) But I'd also rather not mess around with the existing GitHub actions as part of this PR (and it seems the approach we take should be consistent across all actions that modify the changelog?) -- I've opened an issue so we can handle this change in a separate PR https://github.com/heroku/buildpacks-nodejs/issues/820. Hope that works?

Reply:
Easy to implement though https://github.com/heroku/buildpacks-dotnet/pull/2 :)